app/graphql/mutations/test_result/delete.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Mutations
+  module TestResult
+    # Mutation to delete a TestResult
+    class Delete < BaseMutation
+      include Fields
+
+      graphql_name 'deleteTestResults'
+
+      argument :profile_id, ID, required: true
+      field :profile, Types::Profile, null: false
+      collection_field :test_results, Types::TestResult
+
+      def resolve(args = {})
+        profile = scoped_profiles.find(args[:profile_id])
+        test_results = authorized_test_results(args)
+        test_results.map(&:destroy)
```

Comment:
Part of the reason `test_results` is an array is because we have to authorize `:delete?` on it - I didn't see a good way to authorize `:delete?` on an AR::Relation.